l1-contracts/contracts/governance/L2AdminFactory.sol

```diff
@@ -21,12 +22,15 @@ contract L2AdminFactory {
     address[] public requiredRestrictions;
 
     constructor(address[] memory _requiredRestrictions) {
+        _validateZeroAddress(_requiredRestrictions);
         requiredRestrictions = _requiredRestrictions;
     }
 
     /// @notice Deploys a new L2 admin contract.
     /// @return admin The address of the deployed admin contract.
     function deployAdmin(address[] calldata _additionalRestrictions, bytes32 _salt) external returns (address admin) {
```

Comment:
If want to keep `_validateZeroAddress` it is cheaper to use memory cause `_validateZeroAddress` already uses memory
```suggestion
    function deployAdmin(address[] memory _additionalRestrictions, bytes32 _salt) external returns (address admin) {
```